### PR TITLE
Fix responsiveness issue for casa and volunteer data tables

### LINKED ISSUE #5261: Responsive Support For Volunteer and Casa Cases table
## What type(s) of user does this feature affect?
 - volunteers
 - supervisors
 - admins

## Description
Somewhere between desktop and mobile width(ipad size), the [datatables](https://datatables.net/)' pagination controls display vertically.
Make it so the controls display horizontally when the screen is this size.

## Screenshots of current behavior, if any
image

## How to access the QA site
_Login Details:_  
[Link to QA site](https://casa-qa.herokuapp.com/)  

Login Emails: 
- volunteer1@example.com  view site as a volunteer
- supervisor1@example.com view site as a supervisor
- casa_admin1@example.com view site as an admin
- all_casa_admin1@example.com view site as an all casa admin
  - go to `/all_casa_admins/sign_in`  

password for all users: 12345678  

### Questions? Join Slack!

We highly recommend that you join us in slack https://rubyforgood.herokuapp.com/ #casa channel to ask questions quickly and hear about office hours (currently Tuesday 5-7pm Pacific), stakeholder news, and upcoming new issues.

### PR DESCRIPTION
**Pull request by SajjadAhmad14**
### What github issue is this PR for, if any?
Resolves #5261

### What changed, and why?
The pagination for volunteers and casa cases tables is now responsive. 

### How will this affect user permissions?
- Volunteer permissions:
- Supervisor permissions:
- Admin permissions:

### Screenshots please :)

responsive-table

### CODE REVIEW

**Comment by schoork** [comment]
@SajjadAhmad14 Could you give me some steps to get to a page to see this in action? The best page I can find only has two pages.

**Comment by SajjadAhmad14** [comment]
> @SajjadAhmad14 Could you give me some steps to get to a page to see this in action? The best page I can find only has two pages.

@schoork How I approached is by tweaking into the db/seeds files to increase the size of entries for volunteers and casa cases so that I could test the resposiveness for large set of data in the tables.

**Comment by schoork** [comment]
@SajjadAhmad14 I'm still getting the pagination links set vertically. Are you seeing something different?
Screenshot 2023-11-02 at 5 33 18 AM
Screenshot 2023-11-02 at 5 33 12 AM

**Comment by SajjadAhmad14** [comment]
Hi Samuel! Yes the page numbers and buttons(previous and next) on large
screen (>1400px) are showing vertically. You want to be them horizontal on
all screens? Also please confirm that buttons(previous and next) should
also be in one line so that they are also horizontal on all screens? Thanks
for the kind review!

**Comment by schoork** [comment]
Yes, please make all buttons horizontal across all screen sizes. Ticket is a little confusing.

**Comment by SajjadAhmad14** [comment]
Great! I pushed recent changes and now both previous and next buttons are in one line. Also the numbers are responsive on all screens as well.

casa-cases_responsive
volunteers-responsive

**Comment by schoork** [comment]
Wow! Very quick @SajjadAhmad14 . I'm actually a fan of the previous and next buttons being on the same line as the page number buttons, but I'd like to see if we can @bcastillo32 or @MalloryPriceDesign to weigh in.

**Comment by MalloryPriceDesign** [comment]
> Wow! Very quick @SajjadAhmad14 . I'm actually a fan of the previous and next buttons being on the same line as the page number buttons, but I'd like to see if we can @bcastillo32 or @MalloryPriceDesign to weigh in.

@SajjadAhmad14 thanks for working on this! If possible, could you align the previous and next buttons on the same line as the page number buttons?

**Comment by SajjadAhmad14** [comment]
> > Wow! Very quick @SajjadAhmad14 . I'm actually a fan of the previous and next buttons being on the same line as the page number buttons, but I'd like to see if we can @bcastillo32 or @MalloryPriceDesign to weigh in.
> 
> @SajjadAhmad14 thanks for working on this! If possible, could you align the previous and next buttons on the same line as the page number buttons?

Thanks for the review! These buttons can be on the same line but only for desktop screen size. These buttons got hidden on tablet and small sizes if they are on the same as the page numbers. So for medium and small screen sizes I have to adjust them to be on a separate line for the reason of visibilty and this way the pagination style does not look same for all the screen sizes. I think the better user experience is that if we can make it look same for all the screens sizes and that's only possible with the solution that numbers and buttons align on different lines. But I am open for suggestions! Thanks!

File: app/assets/stylesheets/shared/layout.scss [hunk 1]
```diff
@@ -114,3 +114,51 @@ div.row.volunteer-filters {
     }
   }
 }
+
+
+
+@media only screen and (max-width: 2560px) {
+
+  #casa-cases_paginate.dataTables_paginate.paging_simple_numbers {
```
Comment (schoork) [inline app/assets/stylesheets/shared/layout.scss:122]: What's the reason for using the id of the tables in these? CSS won't get propagated to all data tables.

**Review by schoork — CHANGES_REQUESTED** [review]

**Comment by schoork** [comment]
@SajjadAhmad14 We need to find a solution where 

1. all the buttons are on the same line, and
2. the previous and next buttons are shown in some form

Here are some screenshots of [PlainAdmin's pagination|https://demo.plainadmin.com/tables-responsive#0] (we use a lot of PlainAdmin styling in the site). The `>>` and `<<` buttons might be helpful for fitting in the mobile view if the `Previous` and `Next` are too wide.

Happy to jump in/help if you get stuck.
 
Screenshot 2023-11-06 at 7 33 31 AM
Screenshot 2023-11-06 at 7 33 24 AM
Screenshot 2023-11-06 at 7 33 11 AM

File: app/assets/stylesheets/shared/layout.scss (hunk 1 above)
Comment (SajjadAhmad14) [reply app/assets/stylesheets/shared/layout.scss:122]: @schoork How do you propose to target the pagination? I am currently using CSS media queries to target different classes to make the pagination responsive. Can you guide me should I not use the CSS and use some JavaScript instead?

Comment (schoork) [reply app/assets/stylesheets/shared/layout.scss:122]: I just wouldn't use an id here.
```
#casa-cases_paginate.dataTables_paginate.paging_simple_numbers {

.dataTables_paginate.paging_simple_numbers {
```

**Comment by schoork** [comment]
Screenshots for the new changes.
Screenshot 2023-11-18 at 10 40 42 AM
Screenshot 2023-11-18 at 10 40 33 AM
Screenshot 2023-11-18 at 10 40 23 AM

**Comment by schoork** [comment]
FYI. There is currently no way to change the number of numbered page buttons based on the size of the window (mobile does work better with fewer buttons).
https://datatables.net/forums/discussion/73664

**Review by schoork — APPROVED** [review]

**Pull request merged**